awx/main/tasks.py

```diff
@@ -1651,7 +1739,24 @@ class RunProjectUpdate(BaseTask):
 
     @property
     def proot_show_paths(self):
-        return [settings.PROJECTS_ROOT]
+        show_paths = [settings.PROJECTS_ROOT]
+        if self.job_private_data_dir:
+            show_paths.append(self.job_private_data_dir)
+        return show_paths
+
+    def __init__(self, *args, job_private_data_dir=None, **kwargs):
+        super(RunProjectUpdate, self).__init__(*args, **kwargs)
+        self.playbook_new_revision = None
+        self.original_branch = None
+        self.job_private_data_dir = job_private_data_dir
+
+    def event_handler(self, event_data):
+        super(RunProjectUpdate, self).event_handler(event_data)
+        returned_data = event_data.get('event_data', {})
+        if returned_data.get('task_action', '') == 'set_fact':
```

Comment:
Normally I get nervous about custom event handling for performance reasons, but if this sort of thing is restricted to project update events (it seems to be), I think we're okay.

Reply:
Also, nice defensive programming here w/ all the `.get()`s

Reply:
All of this said, I _think_ that runner actually exposes the fact data via another mechanism that might be easier than what you're doing here.

Example: https://github.com/ansible/awx/commit/99478f5d255986cba796b78551a4e2eaf5ef5f17#diff-95c3599b75db21867b3991cee7b89575R272

Reply:
Good points, it definitely crossed my mind. To start from the requirements - I want a simple-as-possible programatic pure-python data pipeline for values to come from my playbook, through runner, to the code that is calling runner. I landed on the event callback, but we can totally go a different direction if a better option presents itself. I don't remember exactly, but I was having some headaches with the bubblewrap / error handling of the old writing to file method.

Reply:
Yep, I'm fine with what's in this PR as-is.